lib/src/main/scala/org/typelevel/scalacoptions/ScalaVersion.scala

```diff
@@ -44,6 +44,18 @@ object ScalaVersion {
   val V3_3_0   = ScalaVersion(3, 3, 0)
   val V3_3_1   = ScalaVersion(3, 3, 1)
 
+  def fromString(version: String): Either[IllegalArgumentException, ScalaVersion] = {
+    val regex = raw"""(\d+)\.(\d+)\.(\d+).*""".r
```

Comment:
I wonder, what is the last part `.*` in the regex is supposed for?
To me it looks like it's going to accept any arbitrary string after the last digit, like "1.2.3abc" – is that correct?

Another thought – perhaps it makes sense to keep the pre-compiled regex value in a private value outside of the method. To avoid getting it compiled every time upon the method call. Wdyt?

Reply:
> but you are right here it's a bug and would blow on the `patch.toLong`

Wait, why is that? The three parenthetical groups in the regex are the major, minor, and patch, the other stuff would be omitted, no?

I hate to say it but I think the answer here is we need to write some tests for this stuff :) I don't see why we shouldn't support prerelease scala versions.

Reply:
I added some tests and support for `-RC` or `-M..` versions. Let me know what you think @armanbilge